src/cljs/rems/status_modal.cljs

```diff
@@ -1,67 +1,118 @@
+  See `rems.status-modal/component` for values to use in the calls."
   (:require [re-frame.core :as rf]
             [reagent.core :as r]
+            [rems.common-util :refer [deep-merge]]
             [rems.guide-functions]
             [rems.modal :as modal]
             [rems.spinner :as spinner]
             [rems.text :refer [text]])
   (:require-macros [rems.guide-macros :refer [component-info example]]))
 
-(defn- status-widget [status error]
-  [:div {:class (when (= :failed status) "alert alert-danger")}
-   (case status
-     nil ""
-     :pending [spinner/big]
-     :saved [:div [:i {:class ["fa fa-check-circle text-success"]}] (text :t.form/success)]
-     :failed [:div [:i {:class "fa fa-times-circle text-danger"}]
-              (str (text :t.form/failed)
-                   (when (:key error)
-                     (str ": " (text (:key error))))
-                   (when-let [text (:status-text error)]
-                     (str ": " text))
-                   (when-let [text (:status error)]
-                     (str " (" text ")")))])])
+(defn- format-errors [errors]
+  [:div (for [error errors]
+          [:p
+           (when (:key error)
+             (str (text (:key error))))
+           (when (:type error)
+             (str (text (:type error))))
+           (when-let [text (:status-text error)]
+             (str text))
+           (when-let [text (:status error)]
+             (str " (" text ")"))])])
+
+(defn- status-widget [success? error-content]
+  (cond
+    (and (not success?) (not error-content)) [spinner/big]
+    success? [:p#status-success [:i {:class ["fa fa-check-circle text-success"]}] (text :t.form/success)]
+    error-content [:div [:p#status-failed [:i {:class "fa fa-times-circle text-danger"}] (text :t.form/failed)]
+                   error-content]))
+
+(rf/reg-event-db ::set-state (fn [db [_ state]] (assoc db ::state state)))
+(rf/reg-event-db ::merge-state (fn [db [_ state]] (update db ::state deep-merge state)))
+(rf/reg-sub ::state (fn [db _] (::state db)))
 
 (defn status-modal
   "Modal component showing the status of an action.
 
-  `:status` - Show spinner while `:pending`. Either `:saved` or `:failed`
-            status is shown with the internal status-widget that
-            may show the `:error` contents.
-  `:description` - the title of the modal.
-  `:content` - additional content to show after the status widget.
-  `:error` - error that may contain :key, :status and :status-text
-           like translated errors or http errors
-  `:on-close` - callback is called when the modal wants to close itself"
-  [{:keys [description status error content on-close]}]
-  [modal/notification {:title description
-                       :content [:div [status-widget status error] content]
-                       :on-close on-close
-                       :shade? true}])
+  `:result`        - Either {:success? true} or {:error ...} {:errors ...}
```

Comment:
what are these keys? keys of `initial-state` and `opts` in the further calls?

Reply:
It's a mistake of updating the docstring, will fix here.